WordPress/Classes/ViewRelated/Blog/Blog Dashboard/Cards/Pages/DashboardPageCell.swift

```diff
@@ -0,0 +1,258 @@
+
+        var icon: UIImage? {
+            switch self {
+            case .published:
+                return UIImage(named: "icon.globe")
+            case .scheduled:
+                return UIImage(named: "icon.calendar")
+            case .draft:
+                return UIImage(named: "icon.verse")
+            }
+        }
+    }
+
+    // MARK: Views
+
+    private lazy var iconImageView: UIImageView = {
+        let imageView = UIImageView()
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+        imageView.tintColor = .secondaryLabel
+        imageView.heightAnchor.constraint(equalToConstant: Metrics.iconImageViewSize).isActive = true
+        imageView.widthAnchor.constraint(equalToConstant: Metrics.iconImageViewSize).isActive = true
+        addSubview(imageView)
+        return imageView
+    }()
+
+    private lazy var titleLabel: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.adjustsFontForContentSizeCategory = true
+        label.adjustsFontSizeToFitWidth = true
+        label.font = WPStyleGuide.regularTextFont()
+        label.numberOfLines = 1
+        label.textColor = .secondaryLabel
+        label.setContentHuggingPriority(.defaultHigh, for: .horizontal)
+        return label
+    }()
+
+    // MARK: Initializers
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        commonInit()
+    }
+
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        commonInit()
+    }
+
+    // MARK: Public Functions
+
+    func configure(for status: BasePost.Status?) {
+        guard let pageStatus = status?.pageStatus else {
+            return
+        }
+        titleLabel.text = pageStatus.title
+        iconImageView.image = pageStatus.icon
+    }
+
+    // MARK: Helpers
+
+    private func commonInit() {
+        applyStyles()
+        setupViews()
+    }
+
+    private func applyStyles() {
+        backgroundColor = .secondarySystemBackground
+        layer.cornerRadius = Metrics.cornerRadius
+    }
+
+    private func setupViews() {
+        addSubviews([iconImageView, titleLabel])
+
+        // Icon Image View Constraints
+        iconImageView.leadingAnchor.constraint(equalTo: leadingAnchor, constant: Metrics.iconLeadingSpace).isActive = true
+        iconImageView.centerYAnchor.constraint(equalTo: centerYAnchor).isActive = true
+
+        // Title label Constraints
+        titleLabel.leadingAnchor.constraint(equalTo: iconImageView.trailingAnchor, constant: Metrics.titleLabelMargins.leading).isActive = true
+        titleLabel.topAnchor.constraint(equalTo: topAnchor, constant: Metrics.titleLabelMargins.top).isActive = true
+        titleLabel.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -Metrics.titleLabelMargins.bottom).isActive = true
+        titleLabel.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -Metrics.titleLabelMargins.trailing).isActive = true
```

Comment:
[nit] WDYT of using NSConstraintLayout.activate instead?

From [Apple docs](https://developer.apple.com/documentation/uikit/nslayoutconstraint/1526955-activate)
> This convenience method provides an easy way to activate a set of constraints with one call. The effect of this method is the same as setting the [isActive](https://developer.apple.com/documentation/uikit/nslayoutconstraint/1527000-isactive) property of each constraint to true. **Typically, using this method is more efficient than activating each constraint individually.**

Reply:
Good point! Updated in 9b0ec64d71942a5c64862b9c5325ad206393d07c 👍